### PR TITLE
Yast2rmt improvements

### PR DESCRIPTION
**Pull request by wstephenson**

### CODE REVIEW

File: src/lib/rmt/wizard_final_page.rb [hunk 1]
```diff
@@ -67,10 +85,38 @@ def back_handler
...
   def run
-    unless (Yast::Service.Enable('rmt-server') && Yast::Service.Restart('rmt-server'))
-      Yast::Report.Error(_("Failed to enable and restart service 'rmt-server'"))
-    end
+    rmt_service_start
     render_content
     event_loop
   end
+
+  def rmt_service_start
```
Comment (thutterer) [inline src/lib/rmt/wizard_final_page.rb]: This should be reorganized a bit, I feel.
As this method (`rmt_service_start`) handles the dialog, does the actual service restarting and conditionally calls another method (`rmt_enable_timers`).
Please remove this nesting. Maybe you can do all of that in in the run method? Or better split out the actual actions into methods (like already done well in `rmt_enable_timers`), but not mix concerns.

And maybe by making this clearer, you also spot why it currently fails (for me) with:
oh_no_yast

I get this after the _success_ message appeared.

**Comment by thutterer** [comment]
OK, besides the timer issue, it looks pretty good now.
One improvement idea: When I press _OK_ on the success pop-up of the service page, this pop-up disappears but the wizard stays on that page. It would be nicer if the wizard just moved on to the next (final) page. Otherwise, the screen reads again like "activating timers ..." and I have to press "Finish" manually. Also, now we have two pages where the _next_ button says _Finish_.
Would be cool if you could improve this, but not mandatory to get this PR through :wink:

**Comment by thutterer** [comment]
Thanks for the small fix in 48b3509 :+1: Still LGTM

**Comment by ikapelyukhin** [comment]
Those bits also need to be in the default config in `database` section:
```
  adapter: mysql2
  encoding: utf8
  timeout: 5000
  pool: 5
```

**Comment by ikapelyukhin** [comment]
@fschueller Aligning the config shouldn't be required anymore (see https://github.com/SUSE/rmt/pull/177).

It should work now with the most minimal config:
```
scc:
  username: user_mcuserface
  password: secret

database:
  username: rmt
  password: rmt
  database: rmt
```

File: src/lib/rmt/utils.rb [hunk 2]
```diff
@@ -33,29 +33,9 @@ class RMT::Utils
       'password' => ''
     },
     'database' => {
-      'host' => 'localhost',
```
Comment (ikapelyukhin) [inline src/lib/rmt/utils.rb]: That might be needed by the Yast module itself when it invokes `mysql` commands.

Comment (ikapelyukhin) [reply src/lib/rmt/utils.rb]: Although I've tested the package and it seems to work fine :thinking:

Comment (ikapelyukhin) [reply src/lib/rmt/utils.rb]: I'm not sure why it works though, I'd vote for bringing `host` back.

Comment (fschueller) [reply src/lib/rmt/utils.rb]: Ok I'll take care of that.

**Comment by wstephenson** [comment]
@thutterer I addressed your UX feedback about the service start page using a popup that goes away when the service has started/failed and update the base page with results - please check

@all note that the feedback popup uses a block and is beyond my rspec-fu to test with an expectation.

File: src/lib/rmt/ssl/certificate_generator.rb [hunk 3]
```diff
@@ -59,6 +59,9 @@ def ca_present?
   end
 
   def server_cert_present?
+    # NB this doesn't check the second file if the first one exists
+    # An improvement would be to look for the absence of any ssl configuration and proceed in that case,
+    # but leave ssl alone if any configuration (hand-edited or incomplete) is
```
Comment (ikapelyukhin) [inline src/lib/rmt/ssl/certificate_generator.rb]: The last sentence incomplete seems, as if by Yoda master written it was. :grimacing:

**Comment by thutterer** [comment]
> @thutterer I addressed your UX feedback about the service start page using a popup that goes away when the service has started/failed and update the base page with results - please check

Hm, okay this is already better. But please add a `TODO` to improve it, since it is a bit of an optimistic cheat, since:

- the dialog can be moved around in the graphical YaST
- and when things fail, you can see a success and error message at the same time

not_really

Good enough for now, but should be improved later on.

**Comment by thutterer** [comment]
@wstephenson 4ef4d6a :1st_place_medal:

**Comment by wstephenson** [comment]
:fireworks:

**Comment by lagartoflojo** [comment]
[LGTM](http://lgtm.in/i/IOsCExkon)

**Pull request merged**